int32 hour
int32 minute
int32 second
---
